### PR TITLE
refactor(server): make `ProxyGetRequestLayer` http middleware support multiple path-method pairs

### PR DESCRIPTION
**Pull request by koushiro**
**BREAKING CHNAGES**:

- change `ProxyGetRequestLayer::new` method to support multiple path-method pairs
- remove `ProxyGetRequest::new` method, which is useless

### CODE REVIEW

**Comment by koushiro** [comment]
@niklasad1 @jsdw PTAL

File: server/Cargo.toml [hunk 1]
```diff
@@ -15,27 +15,28 @@ publish = true
 
 [dependencies]
 futures-util = { version = "0.3.14", default-features = false, features = ["io", "async-await-macro"] }
-jsonrpsee-types = { workspace = true }
```
Comment (niklasad1) [inline server/Cargo.toml:18]: did you sort the dependencies here or why is this changed?

Comment (koushiro) [reply server/Cargo.toml:18]: Yes, this is automatically done by my vscode extension if I change the `Cargo.toml`.

**Review by niklasad1 — APPROVED** [review]
Looks good
I guess it may be useful for users that want to expose several endpoints.

File: server/Cargo.toml (hunk 1 above)
Comment (lexnv) [reply server/Cargo.toml:18]: dq: that sounds useful, what extension are you using?

File: server/src/middleware/http/proxy_get_request.rs [hunk 2]
```diff
@@ -53,29 +55,40 @@ pub struct InvalidPath(String);
 /// See [`ProxyGetRequest`] for more details.
 #[derive(Debug, Clone)]
 pub struct ProxyGetRequestLayer {
-	path: String,
-	method: String,
+	// path => method mapping
+	methods: Arc<FxHashMap<String, String>>,
```
Comment (lexnv) [inline server/src/middleware/http/proxy_get_request.rs:59]: I think a plain `HashMap` would work as well, don't that would bring a noticeable performance hit 
```suggestion
	methods: Arc<HashMap<String, String>>,
```

File: server/src/middleware/http/proxy_get_request.rs [hunk 3]
```diff
@@ -53,29 +55,40 @@ pub struct InvalidPath(String);
...
-		if !path.starts_with('/') {
-			return Err(InvalidPath(path));
+	/// The request `GET /path` is redirected to the provided method.
+	/// Fails if the path does not start with `/`.
+	pub fn new<P, M>(methods: impl IntoIterator<Item = (P, M)>) -> Result<Self, InvalidPath>
+	where
+		P: Into<String>,
+		M: Into<String>,
+	{
+		let methods = methods
```
Comment (lexnv) [inline server/src/middleware/http/proxy_get_request.rs:72]: I believe we can check for path duplicates. Currently, we'll keep in the hashmap just the last entry over which we iterate.

For example, we currently accept:

```rust
[("/health", "system_health"), ("/health", "system2")]
```

This is a misconfig from the user perspective, atm the implementation keeps the `/health` -> `system2` instead of returning an error

**Review by lexnv — APPROVED** [review]
LGTM! Just some minor notes on deduplication of paths for better user errors
Thanks for contributing 🙏

File: server/src/middleware/http/proxy_get_request.rs (hunk 3 above)
Comment (niklasad1) [reply server/src/middleware/http/proxy_get_request.rs:72]: We could also just use a `Vec` here because we don't expect that many endpoints exposed anyway. 
HashMap is probably a bit overkill but either will work...

File: server/Cargo.toml (hunk 1 above)
Comment (koushiro) [reply server/Cargo.toml:18]: https://taplo.tamasfe.dev/

File: server/src/middleware/http/proxy_get_request.rs (hunk 3 above)
Comment (koushiro) [reply server/src/middleware/http/proxy_get_request.rs]: Added check and error type for duplicate path

**Review by lexnv — APPROVED** [review]
Nice 👍
